### PR TITLE
Continous parser api

### PR DESCRIPTION
**Pull request by jhoobergs**

### CODE REVIEW

File: api/controllers/PackageController.js [hunk 1]
```diff
@@ -105,13 +107,87 @@ module.exports = {
...
+    var packageVersionCriteria = {};
+    if(criteria.parser_version)
+      packageVersionCriteria.parser_version = criteria.parser_version;
+    var packageCriteria = criteria;
+    delete packageCriteria.parser_version;
+
+    var queryPromise;
+
+    var isPopularitySort = (sort === 'popularity');
+    if( Object.prototype.toString.call(sort ) === '[object Array]'){
```
Comment (ludov04) [inline api/controllers/PackageController.js]: Can't you use `https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/isArray` to test if it's an array ?

Comment (jhoobergs) [inline api/controllers/PackageController.js]: Yes, it does exactly the same

File: api/controllers/ParserController.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+module.exports = {
+  list: function(req, res) {
+    var limit = Utils.parseLimit(req),
+      offset = Utils.parseSkip(req),
+      parser_version = req.param('parser_version');
+
+    sequelize.query(`SELECT p.name as name, pv.version as version FROM DownloadStatistics ds, Packages p, PackageVersions pv, ParsingJobs job
```
Comment (ludov04) [inline api/controllers/ParserController.js]: This query should be in the models, see others models to see how to add custom queries to it, 
I would even put it in the Packages model, as the main entity that you are retrieving is a Packages.
You can then call Packages.findOldUnfailedPackage or whatever you call it directly from here.
That keeps the database logic out of the controllers

File: config/routes.js [hunk 3]
```diff
@@ -179,6 +179,8 @@ module.exports.routes = {
   'get /api/searchpackages' : 'SearchController.packageSearch',
   'get /api/searchfunctions' : 'SearchController.functionSearch',
   'get /search_packages' : 'SearchController.packageSearch',
-  'get /search_functions' : 'SearchController.functionSearch'
+  'get /search_functions' : 'SearchController.functionSearch',
 
+  // Parser
+  'get /parser/list': 'ParserController.list',
```
Comment (ludov04) [inline config/routes.js]: What you are getting back is json and consumable by other services, so I would put the endpoint under `/api`, something like `/api/packages/to-parse`

**Pull request merged**